README.md

```diff
@@ -117,6 +117,69 @@ node {
 ```
 In the future we might migrate to a [BuildStep](http://javadoc.jenkins-ci.org/hudson/tasks/BuildStep.html) instead of a BuildWrapper.
 
+After integration of the [hashicorp-vault-pipeline-plugin](https://github.com/jenkinsci/hashicorp-vault-pipeline-plugin) this is now also possible
+##### Using global vault configuration
+
+```
+pipeline {
+    agent any
+    environment {
+        SECRET = vault path: 'secrets', key: 'username'
+    }
+    stages {
+        stage("read vault key") {
+            steps {
+                echo "${SECRET}"
+            }
+        }
+    }
+}
+```
+
+##### Using pipeline specific configuration
+
+```
+pipeline {
+    agent any
+    environment {
+        SECRET = vault path: 'secrets', key: 'username', vaultUrl: 'https://my-vault.com:8200', credentialsId: 'my-creds', engineVersion: 2
+    }
+    stages {
+        stage("read vault key") {
+            steps {
+                echo "${SECRET}"
+            }
+        }
+    }
+}
+```
+
+##### Masking secrets in console output
+By default, the plugin does not hide any accidental printing of secret to console. This becomes an issue because `set -x` is set by default in pipeline, so each command with the secrets being passed in will be printed.
+
+[Masked Password Plugin is Required](https://wiki.jenkins.io/display/JENKINS/Mask+Passwords+Plugin)
+
+```
+pipeline {
+    agent any
+    environment {
+        SECRET1    = vault path: 'secrets', key: 'password1', vaultUrl: 'https://my-vault.com:8200', credentialsId: 'my-creds'
+        SECRET2    = vault path: 'secrets', key: 'password2', vaultUrl: 'https://my-vault.com:8200', credentialsId: 'my-creds'
+        NOT_SECRET = vault path: 'secrets', key: 'username', vaultUrl: 'https://my-vault.com:8200', credentialsId: 'my-creds'
+    }
+    stages {
+        stage("read vault key") {
+            steps {
+              wrap([$class: 'MaskPasswordsBuildWrapper', varPasswordPairs: [[password: env['SECRET1'], var: 'SECRET'], [password: env['SECRET2'], var: 'SECRET']]]) {
```

Comment:
@jenkinsci/code-reviewers is there some way in Step class to mark something as masked value?

Reply:
It's something that I would like to have as well, but it was never in the original plugin I implemented here.
As I am going on holiday in a couple of hours, I will not be able to implement this until October at the earliest unfortunately.

Reply:
No worries I can carry it over from here. I was just asking openly before merging 😄

Reply:
Reading up on it, wrap the step with credential binding internally